ruby_connectors/lib/ruby_connectors/duckdb_connector/reader.rb

```
@@ -0,0 +1,95 @@
+
+            {
+              data: execute_query(conn, paginated_query),
+              offset: effective_offset,
+              limit: effective_limit,
+              total_records: get_total_records(conn, base_query)
+            }
+          end
+        end
+      rescue DuckDB::Error => e
+        raise ReadError, "DuckDB read error: #{e.message}"
+      end
```

RuboCop (1.69.1)

[convention] Assignment Branch Condition size for read is too high. [<14, 14, 9> 21.75/17]

(Metrics/AbcSize)